pkg/cluster/cluster.go

```diff
@@ -311,10 +311,11 @@ func isSpecEqual(s1, s2 api.ClusterSpec) bool {
 
 func (c *Cluster) startSeedMember() error {
 	m := &etcdutil.Member{
-		Name:         k8sutil.UniqueMemberName(c.cluster.Name),
-		Namespace:    c.cluster.Namespace,
-		SecurePeer:   c.isSecurePeer(),
-		SecureClient: c.isSecureClient(),
+		Name:          k8sutil.UniqueMemberName(c.cluster.Name),
+		Namespace:     c.cluster.Namespace,
+		SecurePeer:    c.isSecurePeer(),
+		SecureClient:  c.isSecureClient(),
+		ClusterDomain: c.cluster.Spec.Pod.ClusterDomain,
```

Comment:
Check if `c.cluster.Spec.Pod` is `nil` before setting this field. Otherwise we get a segfault:

```console
time="2019-05-09T20:39:40Z" level=info msg="etcd-operator Version: 0.9.4+git"
time="2019-05-09T20:39:40Z" level=info msg="Git SHA: 8d9b1e5d"
time="2019-05-09T20:39:40Z" level=info msg="Go Version: go1.10"
time="2019-05-09T20:39:40Z" level=info msg="Go OS/Arch: linux/amd64"
time="2019-05-09T20:39:40Z" level=info msg="Event(v1.ObjectReference{Kind:\"Endpoints\", Namespace:\"e2e-etcd-operator-e2e-pr-3371\", Name:\"etcd-operator\", UID:\"91df3254-729a-11e9-8ce3-42010a800271\", APIVersion:\"v1\", ResourceVersion:\"111408674\", FieldPath:\"\"}): type: 'Normal' reason: 'LeaderElection' etcd-operator became leader"
time="2019-05-09T20:39:45Z" level=info msg="creating cluster with Spec:" cluster-name=test-etcd-lntbc cluster-namespace=e2e-etcd-operator-e2e-pr-3371 pkg=cluster
time="2019-05-09T20:39:45Z" level=info msg="{" cluster-name=test-etcd-lntbc cluster-namespace=e2e-etcd-operator-e2e-pr-3371 pkg=cluster
time="2019-05-09T20:39:45Z" level=info msg="    \"size\": 3," cluster-name=test-etcd-lntbc cluster-namespace=e2e-etcd-operator-e2e-pr-3371 pkg=cluster
time="2019-05-09T20:39:45Z" level=info msg="    \"repository\": \"quay.io/coreos/etcd\"," cluster-name=test-etcd-lntbc cluster-namespace=e2e-etcd-operator-e2e-pr-3371 pkg=cluster
time="2019-05-09T20:39:45Z" level=info msg="    \"version\": \"3.2.13\"" cluster-name=test-etcd-lntbc cluster-namespace=e2e-etcd-operator-e2e-pr-3371 pkg=cluster
time="2019-05-09T20:39:45Z" level=info msg="}" cluster-name=test-etcd-lntbc cluster-namespace=e2e-etcd-operator-e2e-pr-3371 pkg=cluster
panic: runtime error: invalid memory address or nil pointer dereference
[signal SIGSEGV: segmentation violation code=0x1 addr=0x98 pc=0x10c9589]

goroutine 73 [running]:
github.com/coreos/etcd-operator/pkg/cluster.(*Cluster).startSeedMember(0xc4202345a0, 0xc42048fe00, 0x30)
	/go/src/github.com/coreos/etcd-operator/pkg/cluster/cluster.go:318 +0x149
github.com/coreos/etcd-operator/pkg/cluster.(*Cluster).bootstrap(0xc4202345a0, 0x0, 0x3)
	/go/src/github.com/coreos/etcd-operator/pkg/cluster/cluster.go:344 +0x2b
github.com/coreos/etcd-operator/pkg/cluster.(*Cluster).prepareSeedMember(0xc4202345a0, 0x0, 0x0)
	/go/src/github.com/coreos/etcd-operator/pkg/cluster/cluster.go:167 +0x5a
github.com/coreos/etcd-operator/pkg/cluster.(*Cluster).create(0xc4202345a0, 0xc420472f60, 0x10d95a7)
	/go/src/github.com/coreos/etcd-operator/pkg/cluster/cluster.go:161 +0x111
github.com/coreos/etcd-operator/pkg/cluster.(*Cluster).setup(0xc4202345a0, 0xc4202bfa70, 0xc420472f80)
	/go/src/github.com/coreos/etcd-operator/pkg/cluster/cluster.go:148 +0x83
github.com/coreos/etcd-operator/pkg/cluster.New.func1(0xc4202345a0)
	/go/src/github.com/coreos/etcd-operator/pkg/cluster/cluster.go:105 +0x35
created by github.com/coreos/etcd-operator/pkg/cluster.New
	/go/src/github.com/coreos/etcd-operator/pkg/cluster/cluster.go:104 +0x33b
```